snuba/query/processors/physical/column_filter_processor.py

```diff
@@ -12,7 +13,15 @@ class ColumnFilterProcessor(ClickhouseQueryProcessor):
     """
 
     def __init__(self, column_filters: Sequence[str]) -> None:
-        self.__column_filters = column_filters
+        self.__column_filters = set(column_filters)
 
     def process_query(self, query: Query, query_settings: QuerySettings) -> None:
-        pass
+        visitor = ColumnVisitor()
+        condition = query.get_condition()
+        assert (
+            condition is not None
+        ), "Not a valid DELETE query. Delete queries should be in the form of DELETE FROM [db.]table WHERE expr"
+        column_names = condition.accept(visitor)
+        assert (
+            column_names == self.__column_filters
```

Comment:
add the column filters in the message as well. Also, throw a proper exception. asserts can be [optimized away](https://docs.python.org/3/reference/simple_stmts.html#assert) depending on how python is running. Asserts should only be used to assert behavior that should never happen. This can happen

Reply:
Yeah we can use `InvalidQueryException`. I also think we can simplify the error message: `f"Query columns {column_names} must match columns {self.__column_filters}"`